pkg/aws/ec2.go

```diff
@@ -1249,3 +1250,24 @@ func (e EC2Client) DescribeInstanceTypes(client Client) ([]string, error) {
 	}
 	return instanceTypeList, nil
 }
+
+//$ aws ec2 describe-images --filters Name=image-id,Values=ami-01288945bd24ed49a --query "Images[*].Architecture"
+func (e EC2Client) DescribeAMIArchitecture(client Client, amiID string) (string, error) {
+	var amiArchitecture string
+	ec2svc := client.EC2Service.Client
```

Comment:
Struct itself is a client, but why do you need to pass client as  parameter??

Reply:
thank you for comment. i fixed this part.